src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,77 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int INITIAL_INDEX = 0;
+    private K key;
+    private V value;
+    private StorageImpl<K, V>[] arrayElements;
```

Comment:
```suggestion
    private K[] keys;
    private V[] values;
```

You don't need to store the object itself. It is enough to make an array of keys and values.